caddyhttp/fastcgi/fastcgi.go

```diff
@@ -302,6 +308,7 @@ func (h Handler) buildEnv(r *http.Request, rule Rule, fpath string) (map[string]
 		"REMOTE_IDENT":      "", // Not used
 		"REMOTE_USER":       remoteUser,
 		"REQUEST_METHOD":    r.Method,
+		"REQUEST_SCHEME":    requestScheme,
```

Comment:
You don't need a separate function or block, use `r.URL.Scheme`

Reply:
I thought about it, but checked another caddy code for getting scheme and did it similiar.

https://github.com/mholt/caddy/blob/95514da91bbf134baaa6b18e3e9037a5dd165d1c/caddytls/httphandler.go#L56

I am not sure that you always have scheme in url.

Reply:
Should i rewrite this to use r.URL.Scheme?

Reply:
What you have done is the safest but from my experience the url scheme is also accurate. The ACME challenge may be a special case so I'll wait for @mholt and others on this.

Reply:
From the godoc on net/http.Request's URL field:

```
        // For server requests the URL is parsed from the URI
        // supplied on the Request-Line as stored in RequestURI.  For
        // most requests, fields other than Path and RawQuery will be
        // empty. (See RFC 2616, Section 5.1.2)
```

So checking r.TLS is the correct way to test for HTTPS.